doc/Architecture-overview/index.md

```diff
@@ -2,7 +2,158 @@
+
+The Kaa IoT platform consists of the *Kaa server*, *Kaa extensions*, and *endpoint SDKs*.
+
+* The Kaa server implements the back-end part of the platform, performs tenants, applications, users, and devices management, exposes integration interfaces, and offers administrative capabilities.
+* Kaa extensions are independent software modules that improve the platform functionality.
+(In this version of the documentation you will notice some extensions actually managed within the core platform code base. Those are planned to be fully decoupled in future Kaa releases.)
+* Endpoint SDK is a library which provides client-side APIs for the various [Kaa platform features]({{root_url}}Programming-guide/Key-platform-features/) and handles communication, data marshalling, persistence, etc.
+Kaa SDKs are designed to facilitate the creation of client applications to be run on various connected devices - however, client applications that do not use Kaa endpoint SDK are also possible.
+Several implementations of the [Endpoint SDK]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/) are available in different programming languages.
+
+*Kaa cluster* consists of Kaa server nodes that use [Apache ZooKeeper](https://zookeeper.apache.org/) for services coordination.
+Kaa cluster also requires NoSQL and SQL database instances to store endpoint data and metadata, accordingly.
+
+<img src="attach/high-level-architecture.png" width="921" height="1002" style="display: block;margin-left: auto;margin-right: auto;">
+
+# Kaa cluster
+
+Kaa nodes in a cluster run a combination of Control, Operations, and Bootstrap services.
+A system administrator can enable or disable any of the services on a particular node, thus flexibly configuring the cluster deployment (see [general configuration]({{root_url}}Administration-guide/System-Configuration/General-configuration/) for more details).
+
+## Control service
+
+Kaa Control service is responsible for managing overall system data, processing [API calls]({{root_url}}Programming-guide/Server-REST-APIs/) from the web UI and external integrated systems, and delivering corresponding notifications to Operations services.
+Control service maintains an up-to-date list of available Operations services by continuously receiving this information from ZooKeeper.
+Additionally, Control service runs embedded Administrative web UI component, which uses Control service APIs to provide platform users with a convenient web-based interface for managing tenants, user accounts, applications, application data, etc.
+
+To support high availability, a Kaa cluster must include at least two nodes with Control service enabled.
+In HA configuration one of the Control services acts as active and other(s) function in standby mode.
+In case of the active Control service failure, ZooKeeper notifies one of the standby Control service and promotes it to the active Control service.
+
+## Bootstrap service
+
```

Comment:
I think it would be better to switch an order in which these services are described. The Operation service should go first.
Just like they were introduced:
Kaa nodes in a cluster run a combination of Control, Operations, and Bootstrap services